gift_compassion/models/sponsorship_gift.py

```diff
@@ -516,6 +517,18 @@ def action_verify(self):
         return True
 
     @api.multi
+    def action_in_progress(self):
+        self.write({'state': 'In Progress'})
+        self.mapped('message_id').write({'state': 'pending'})
+        return True
+
+    @api.multi
+    def action_suspended(self):
+        self.write({'state': 'suspended'})
+        self.mapped('message_id').write({'state': 'postponed'})
```

Comment:
I don't think you should change the state of the message. It's already done elsewhere